File: dock_spawn/containers/fill_dock_container.dart

class FillDockContainer implements IDockContainer {
  String containerType = "fill";
  DivElement element;
  DockManager dockManager;
  TabHost tabHost;
  int tabOrientation = TabHost.DIRECTION_BOTTOM;
  int get minimumAllowedChildNodes() { return 2; }
  String name;
  
  FillDockContainer([int tabStripDirection = TabHost.DIRECTION_BOTTOM]) {
    this.tabOrientation = tabStripDirection;
    name = getNextId("fill_");
    element = new DivElement();
    element.classes.add("dock-container");
    element.classes.add("dock-container-fill");
    
    tabHost = new TabHost(tabStripDirection: tabOrientation);
    element.nodes.add(tabHost.hostElement);
  }

  
  void setActiveChild(IDockContainer child) {
    tabHost.setActiveTab(child);
  }
  
  void resize(int _width, int _height) {
    element.style.width = "${_width}px";
    element.style.height = "${_height}px";
    tabHost.resize(_width, _height);
  }
  
  void performLayout(List<IDockContainer> children) {
    tabHost.performLayout(children);
  }
  
  void destroy() {
    element.remove();
    element = null;
  }

  void saveState(Map<String, Object> state) {
    state['width'] = width;
    state['height'] = height;
  }
  
  void loadState(Map<String, Object> state) {
    width = state['width'];
    height = state['height'];
  }
  
  Element get containerElement() {
    return element;
  }
  
  int get width() {
    return element.$dom_clientWidth;
  }
  void set width(int value) {
    element.style.width = "${value}px";
    tabHost.resize(value, height);
  }

  int get height() {
    return element.$dom_clientHeight;
  }
  void set height(int value) {
    element.style.height = "${value}px";
    tabHost.resize(width, value);
  }
  

}

File: js/lib/containers/FillDockContainer.js
import "../tab/TabHost";

dockspawn.FillDockContainer = function(dockManager, tabStripDirection)
{
    if (arguments.length == 0)
        return;

    if (tabStripDirection === undefined)
        tabStripDirection = dockspawn.TabHost.DIRECTION_BOTTOM;

    this.dockManager = dockManager;
    this.tabOrientation = tabStripDirection;
    this.name = getNextId("fill_");
    this.element = document.createElement("div");
    this.containerElement = this.element;
    this.containerType = "fill";
    this.minimumAllowedChildNodes = 2;
    this.element.classList.add("dock-container");
    this.element.classList.add("dock-container-fill");
    this.tabHost = new dockspawn.TabHost(this.tabOrientation);
    this.element.appendChild(this.tabHost.hostElement);
}

dockspawn.FillDockContainer.prototype.setActiveChild = function(child)
{
    this.tabHost.setActiveTab(child);
};

dockspawn.FillDockContainer.prototype.resize = function(width, height)
{
    this.element.style.width = width + "px";
    this.element.style.height = height + "px";
    this.tabHost.resize(width, height);
};

dockspawn.FillDockContainer.prototype.performLayout = function(children)
{
    this.tabHost.performLayout(children);
};

dockspawn.FillDockContainer.prototype.destroy = function()
{
    if (removeNode(this.element))
        delete this.element;
};

dockspawn.FillDockContainer.prototype.saveState = function(state)
{
    state.width = this.width;
    state.height = this.height;
};

dockspawn.FillDockContainer.prototype.loadState = function(state)
{
    this.width = state.width;
    this.height = state.height;
};

Object.defineProperty(dockspawn.FillDockContainer.prototype, "width", {
    get: function() { return this.element.clientWidth; },
    set: function(value) { this.element.style.width = value + "px" }
});

Object.defineProperty(dockspawn.FillDockContainer.prototype, "height", {
    get: function() { return this.element.clientHeight; },
    set: function(value) { this.element.style.height = value + "px" }
});
